File: csharp/diagonal_negativos.cs
using System;

namespace diagonal_negativos
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int i, j, negativos;

            Console.Write("Qual a ordem da matriz? ");
            int n = int.Parse(Console.ReadLine());

            int[,] mat = new int[n, n];

            negativos = 0;

            for (i = 0; i < n; i++)
            {
                for (j = 0; j < n; j++)
                {
                    Console.Write("Elemento [" + i + "," + j + "]: ");
                    mat[i,j] = int.Parse(Console.ReadLine());

                    if (mat[i,j] < 0)
                    {
                        negativos = negativos + 1;
                    }
                }

            }

            Console.WriteLine("DIAGONAL PRINCIPAL: ");
            for (i = 0; i < n; ++i)
            {
                Console.Write(mat[i,i] + " ");
            }
            Console.WriteLine();
            Console.Write("QUANTIDADE DE NEGATIVOS = " + negativos);

        }
    }
}

File: java/diagonal_negativos.java
import java.util.Locale;
import java.util.Scanner;

public class diagonal_negativos {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Qual a ordem da matriz? ");
		int n = sc.nextInt();
		
		int negativos;
		
		int mat[][] = new int[n][n];
		
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < n; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				mat[i][j] = sc.nextInt();
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL: ");
		for (int i = 0; i < n; i++) {
				System.out.print(mat[i][i] + " ");
		}
		
		negativos = 0;
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < n; j++) {
				if (mat[i][j] < 0)
				negativos++;
			}
		}
		
		System.out.println();
		System.out.println("QUANTIDADE DE NEGATIVOS = " + negativos);
		
	}

}
